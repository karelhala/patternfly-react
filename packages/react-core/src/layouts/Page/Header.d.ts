import { SFC, HTMLProps, ReactNode } from 'react';
import { OneOf } from '../../typeUtils';

export interface HeaderSidebarProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
  className?: string;
}

export declare const HeaderSidebar: SFC<HeaderSidebarProps>;

export interface HeaderMainProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
  className?: string;
}

export declare const HeaderMain: SFC<HeaderMainProps>;
