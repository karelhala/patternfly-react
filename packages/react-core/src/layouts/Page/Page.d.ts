import { SFC, HTMLProps, ReactNode } from 'react';
import { OneOf } from '../../typeUtils';

export interface PageProps extends HTMLProps<HTMLDivElement> {
  header?: React.ReactNode;
  sidebar?: React.ReactNode;
  children?: React.ReactNode;
  className?: string;
}

declare const Page: SFC<PageProps>;

export default Page;
