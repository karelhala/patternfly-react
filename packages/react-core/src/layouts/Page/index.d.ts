export { default as Page, PageProps } from './Page';
export {
  HeaderSidebar as PageHeaderSidebar,
  HeaderSidebarProps as PageHeaderSidebarProps,
  HeaderMainProps as PageHeaderMainProps,
  HeaderMain as PageHeaderMain
} from './Header';