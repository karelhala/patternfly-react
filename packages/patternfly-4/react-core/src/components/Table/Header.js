import React from 'react';
import { Header } from 'reactabular-table';
import PropTypes from 'prop-types';
import { TableContext } from './Table';
import { isEqual } from 'lodash';
import { mapHeader } from './utils';

const propTypes = {
  className: PropTypes.string,
  headerRows: PropTypes.array
};

const defaultProps = {
  className: ''
};

class ContextHeader extends React.Component {
  shouldComponentUpdate(nextProps) {
    return !isEqual(nextProps.headerRows, this.props.headerRows);
  }

  calculateColumns() {
    const { headerRows } = this.props;
    return headerRows && headerRows.map(oneCol => ({
      ...mapHeader(oneCol),
      ...typeof oneCol !== 'string' && oneCol
    })
    );
  }

  componentDidUpdate() {
    this.props.updateHeaderData && this.props.updateHeaderData(this.calculateColumns());
  }

  componentDidMount() {
    this.props.updateHeaderData && this.props.updateHeaderData(this.calculateColumns());
  }

  render() {
    const { className, headerRows, updateHeaderData, ...props } = this.props;

    return (
      <Header headerRows={[this.calculateColumns()]} {...props} className={className} />
    );
  }
}

const TableHeader = ({ headerData, ...props }) => (
  <TableContext.Consumer>
    {({ updateHeaderData }) => (
      <ContextHeader {...props} updateHeaderData={updateHeaderData} />
    )}
  </TableContext.Consumer>
);

TableHeader.propTypes = propTypes;
TableHeader.defaultProps = defaultProps;

export default TableHeader;
