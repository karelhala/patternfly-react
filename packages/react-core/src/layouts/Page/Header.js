import React from 'react';
import styles from '@patternfly/patternfly-next/layouts/Page/styles.css';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const Header = ({ children, className, ...props }) => (
  <header
    {...props}
    role="banner"
    className={css(styles.pageHeader, className)}
  >
    {children}
  </header>
);

export default Header;

export const HeaderSidebar = ({ children, className, ...props }) => (
  <header {...props} className={css(styles.pageHeaderSideBar, className)}>
    {children}
  </header>
);

export const HeaderMain = ({ children, className, ...props }) => (
  <header {...props} className={css(styles.pageHeaderSideBar, className)}>
    {children}
  </header>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

HeaderSidebar.propTypes = propTypes;
HeaderSidebar.defaultProps = defaultProps;

HeaderMain.propTypes = propTypes;
HeaderMain.defaultProps = defaultProps;
