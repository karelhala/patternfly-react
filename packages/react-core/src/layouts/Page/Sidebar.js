import React from 'react';
import styles from '@patternfly/patternfly-next/layouts/Page/styles.css';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const Sidebar = ({ className, children, ...props }) => (
  <div className={css(styles.pageSidebar, className)}>{children}</div>
);

Sidebar.propTypes = propTypes;
Sidebar.defaultProps = defaultProps;

export default Sidebar;
