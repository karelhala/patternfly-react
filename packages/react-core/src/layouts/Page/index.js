export { default as Page } from './Page';
export { HeaderSidebar, HeaderMain } from './Header';
