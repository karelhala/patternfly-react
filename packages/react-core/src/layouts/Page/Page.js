import React from 'react';
import styles from '@patternfly/patternfly-next/layouts/Page/styles.css';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import Sidebar from './Sidebar';
import Header from './Header';

const propTypes = {
  header: PropTypes.node,
  sidebar: PropTypes.node,
  children: PropTypes.node,
  className: PropTypes.string
};

const defaultProps = {
  header: null,
  sidebar: null,
  children: null,
  className: ''
};

const Page = ({ header, sidebar, children, className, ...props }) => (
  <div {...props} className={css(styles.page, className)}>
    {header && <Header>{header}</Header>}
    {sidebar && <Sidebar>{sidebar}</Sidebar>}
    <main role="main" className={css(styles.main)}>
      {children}
    </main>
  </div>
);

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
