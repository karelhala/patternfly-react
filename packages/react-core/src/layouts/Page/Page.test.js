import React from 'react';
import { mount } from 'enzyme';
import Page from './Page';
import { HeaderMain, HeaderSidebar } from './Header';

test('replace with useful test name', () => {
  const view = mount(
    <Page
      header={
        <React.Fragment>
          <HeaderMain />
          <HeaderSidebar />
        </React.Fragment>
      }
      sidebar="Nav"
    >
      Page Layout Content
    </Page>
  );
  expect(view).toMatchSnapshot();
});
