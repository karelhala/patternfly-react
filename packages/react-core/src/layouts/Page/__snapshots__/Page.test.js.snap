// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace with useful test name 1`] = `
.pf-l-page__header {
  display: grid;
  grid-area: header;
  grid-template-columns: 15rem 1fr;
  grid-template-areas: "header-left header-right";
}
.pf-l-page__sidebar {
  display: block;
}
.pf-l-page {
  display: grid;
  padding: 0px;
  margin: 0px;
  grid-template-columns: 15rem 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: "header header" "main main";
}

<Page
  className=""
  header={
    <UNDEFINED>
      <HeaderMain
        className=""
      />
      <HeaderSidebar
        className=""
      />
    </UNDEFINED>
  }
  sidebar="Nav"
>
  <div
    className="pf-l-page"
  >
    <Header
      className=""
    >
      <header
        className="pf-l-page__header"
        role="banner"
      >
        <HeaderMain
          className=""
        >
          <header
            className=""
          />
        </HeaderMain>
        <HeaderSidebar
          className=""
        >
          <header
            className=""
          />
        </HeaderSidebar>
      </header>
    </Header>
    <Sidebar
      className=""
    >
      <div
        className="pf-l-page__sidebar"
      >
        Nav
      </div>
    </Sidebar>
    <main
      className=""
      role="main"
    >
      Page Layout Content
    </main>
  </div>
</Page>
`;
